node/libs/crypto/src/bn254/error.rs

```diff
@@ -1,5 +1,5 @@
 /// Error type for generating and interacting with bn254.
-#[derive(Debug, thiserror::Error)]
+#[derive(Debug, PartialEq, thiserror::Error)]
```

Comment:
Since derived traits is a part of the public interface of types, deriving `PartialEq` for errors is usually a bad idea; it could be difficult to maintain when new error variants are added. To match errors in tests, you could use `assert!(matches!(_))`, or the [`assert_matches`](https://docs.rs/assert_matches/latest/assert_matches/) crate (the latter produces more informative panic messages).

Reply:
Agree. I refrained from the former due to lack of descriptive panics, but the latter does provides that. 
It's not identical though—it expects a pattern and not an expression for the 2nd leg, but if combined with `assert_eq!` for inner comparisons (assuming that you don't want to just save the expression result in a variable), it continues to provide descriptive panics. 

Changes here: https://github.com/matter-labs/era-consensus/pull/39/commits/ef8cc5f352a9e2243f186cf4d4a69167963d899d